File: programmers/DFS-BFS/네트워크.py
def dfs(x,adj,visited):
    visited[x]=1
    for nx in adj[x]:
        if visited[nx]==0:
            dfs(nx,adj,visited)
def solution(n, computers):
    answer = 0
    adj=[[] for i in range(n)]
    visited=[0]*(n)
    for i in range(n):
        for j in range(n):
            if i!=j and computers[i][j]==1:    
                adj[i].append(j)
    print(adj)
    for i in range(n):
        if visited[i]==0:
            dfs(i,adj,visited)
            answer+=1
    return answer

File: javascript/programmers/DFS_BFS/네트워크.js
function solution(n, computers) {
    var answer = 0;
    let arr = Array.from({ length: n }, () => 0);

    function dfs(x) {
        arr[x] = 1;
        for (let i = 0; i < n; i++) {
            if (arr[i] === 0 && computers[x][i] === 1) {
                dfs(i);
            }
        }
    }

    for (let i = 0; i < n; i++) {
        if (arr[i] === 0) {
            dfs(i);
            answer++;
        }
    }
    return answer;
}